### PR TITLE
Policies feedback (v2)

### PR DESCRIPTION
**Pull request by btry**
refactor post_updateItem by splitting its content and reordering its actions

### Checklist

Please check if your PR fulfills the following specifications:

- [ ] Tests for the changes have been added
- [ ] Docs have been added/updated

### References

Closes #N/A

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -31,7 +31,7 @@ before_script:
 - "./tests/before_script.sh"
 script:
   - vendor/bin/atoum -ft -bf tests/bootstrap.php -d tests/suite-install -ncc
-  - php scripts/mqtt.php --tests > /dev/null 2>&1 &
+  - #php scripts/mqtt.php --tests > /dev/null 2>&1 &
```
Comment (btry) [inline .travis.yml]: Still not used yet. I think I'll use the daemon with DI (see toe PR in experimentation)

File: inc/agent.class.php [hunk 2]
```diff
@@ -306,27 +315,23 @@ public static function showForFleet(PluginFlyvemdmFleet $item) {
...
+      $number = count($rows);
 
       // get the pager
       $pager = Html::printAjaxPager(self::getTypeName(1), $start, $number, '', false);
-      $pager = ''; // disabled because the results are not paged yet
-
-      // get items
-      $agent = new static();
-      $condition = "`$itemFk` = '$items_id' " . $dbUtils->getEntitiesRestrictRequest();
-      $rows = $agent->find($condition, '', '');
```
Comment (btry) [inline inc/agent.class.php]: We already got the items with the previous request used to count them. The template filters the rows for paging, in a similar way as GLPI does currently.

File: inc/policybase.class.php [hunk 3]
```diff
@@ -59,6 +59,34 @@
...
+     */
+   public static final function getEnumBaseTaskStatus() {
+      return [
+         'pending'   => __('Pending', 'flyvemdm'),
+         'received'  => __('Received', 'flyvemdm'),
+         'done'      => __('Done', 'flyvemdm'),
+         'failed'    => __('Failed', 'flyvemdm'),
+         'canceled'  => __('Canceled', 'flyvemdm'),
+      ];
+   }
```
Comment (btry) [inline inc/policybase.class.php]: This array of status is common to all policies.

File: inc/policybase.class.php [hunk 4]
```diff
@@ -59,6 +59,34 @@
...
+     * @return array
+     */
+   public static final function getEnumBaseTaskStatus() {
+      return [
+         'pending'   => __('Pending', 'flyvemdm'),
+         'received'  => __('Received', 'flyvemdm'),
+         'done'      => __('Done', 'flyvemdm'),
+         'failed'    => __('Failed', 'flyvemdm'),
+         'canceled'  => __('Canceled', 'flyvemdm'),
+      ];
```
Comment (btry) [inline inc/policybase.class.php]: @rafaelje: here is the list of status common to all policies
- pending => not sent yet to the agent
- received => the agent acknoledge recepit of the task
- done => the agent completed the task (successfully)
- failed => the agent tried and failed to complet ethe task
- canceled => the backend canceled the task (and set this status itself) The agent should not set we don't have use case for that, but why not)

File: inc/policydeployapplication.class.php [hunk 5]
```diff
@@ -241,4 +241,10 @@ public function showValue(PluginFlyvemdmTask $task) {
       }
       return NOT_AVAILABLE;
    }
+
+   public static function getEnumSpecificStatus() {
+      return [
+         'waiting' => __('Waiting', 'flyvemdm'),
+      ];
+   }
```
Comment (btry) [inline inc/policydeployapplication.class.php]: @rafaelje: this status is for appp / file deployment and removal. If you need a user action, you may send this status to the backend to tell you need to wait for something.

File: inc/policybase.class.php (hunk 4 above)
Comment (rafaelje) [reply inc/policybase.class.php]: I'll implement this in the Android Agent, now the agent has generic response

File: inc/policydeployapplication.class.php (hunk 5 above)
Comment (rafaelje) [reply inc/policydeployapplication.class.php]: Fine

**Pull request merged**